python/private/py_wheel.bzl

```diff
@@ -203,6 +203,427 @@ _DESCRIPTION_FILE_EXTENSION_TO_TYPE = {
+            return True
+        else:
+            contexts.pop()
+            return False
+
+    def accept_prerelease():
+        """PEP 440: Pre-releases."""
+        context = open_context(contexts[-1]["start"])
+
+        # PEP 440: Pre-release separators
+        accept(in_(["-", "_", "."]), "")
+
+        if not accept_pre_l():
+            contexts.pop()
+            return False
+
+        accept(in_(["-", "_", "."]), "")
+
+        if not accept_digits():
+            # PEP 440: Implicit pre-release number
+            context["norm"] += "0"
+
+        close_context()
+        return True
+
+    def accept_implicit_postrelease():
+        """PEP 440: Implicit post releases."""
+        context = open_context(contexts[-1]["start"])
+
+        if accept(is_("-"), "") and accept_digits():
+            context["norm"] = ".post" + context["norm"]
+            close_context()
+            return True
+
+        contexts.pop()
+        return False
+
+    def accept_explicit_postrelease():
+        """PEP 440: Post-releases."""
+        context = open_context(contexts[-1]["start"])
+
+        # PEP 440: Post release separators
+        if not accept(in_(["-", "_", "."]), "."):
+            context["norm"] += "."
+
+        # PEP 440: Post release spelling
+        if (
+            accept_string("post", "post") or
+            accept_string("rev", "post") or
+            accept_string("r", "post")
+        ):
+            accept(in_(["-", "_", "."]), "")
+
+            if not accept_digits():
+                # PEP 440: Implicit post release number
+                context["norm"] += "0"
+
+            close_context()
+            return True
+
+        contexts.pop()
+        return False
+
+    def accept_postrelease():
+        """PEP 440: Post-releases."""
+        open_context(contexts[-1]["start"])
+
+        if accept_implicit_postrelease() or accept_explicit_postrelease():
+            close_context()
+            return True
+
+        contexts.pop()
+        return False
+
+    def accept_devrelease():
+        """PEP 440: Developmental releases."""
+        context = open_context(contexts[-1]["start"])
+
+        # PEP 440: Development release separators
+        if not accept(in_(["-", "_", "."]), "."):
+            context["norm"] += "."
+
+        if accept_string("dev", "dev"):
+            accept(in_(["-", "_", "."]), "")
+
+            if not accept_digits():
+                # PEP 440: Implicit development release number
+                context["norm"] += "0"
+
+            close_context()
+            return True
+
+        contexts.pop()
+        return False
+
+    def accept_local():
+        """PEP 440: Local version identifiers."""
+        open_context(contexts[-1]["start"])
+
+        if accept(is_("+"), "+") and accept_alnum():
+            accept_separator_alnum_sequence()
+            close_context()
+            return True
+
+        contexts.pop()
+        return False
+
+    open_context(0)
```

Comment:
It would be better to initialize the `contexts` list with the right first entry to begin with, because `contexts = []` is not a valid list when used in the parsing functions.

Reply:
Done.